server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileProcessor.java

```diff
@@ -746,7 +747,8 @@ public void syncFlush() throws IOException {
     synchronized (tmpMemTable) {
       try {
         long startWait = System.currentTimeMillis();
-        while (flushingMemTables.contains(tmpMemTable)) {
+        HashSet<IMemTable> flushingMemSet = new HashSet<>(flushingMemTables);
+        while (flushingMemSet.contains(tmpMemTable)) {
```

Comment:
I think the change here may cause concurrent issues: `flushingMemTables` may be changed by other working threads.

Reply:
Ditto has not modeled the semantics of multi-threading. It is true that the transformation here is not sound. Ditto just finds that there exists a performance issue here. The method contains of a list is quite inefficient. Moreover, it is invoked frequently in this loop.

Reply:
Theoretically speaking, the statement `flushingMemSet.contains(tmpMemTable)` is inefficient, so at least your finding is correct. 😊